homeassistant/components/konnected/config_flow.py

```diff
@@ -267,9 +267,23 @@ async def async_step_ssdp(self, discovery_info):
         else:
             # extract host/port from ssdp_location
             netloc = urlparse(discovery_info["ssdp_location"]).netloc.split(":")
-            return await self.async_step_user(
-                user_input={CONF_HOST: netloc[0], CONF_PORT: int(netloc[1])}
-            )
+            try:
+                status = await get_status(self.hass, netloc[0], int(netloc[1]))
+            except CannotConnect:
+                return self.async_abort(reason="connection_error")
```

Comment:
Please use an existing abort reason string or add the new reason to strings.json.

Reply:
Ok, will do.

Reply:
Consider adding a 

`self._async_abort_entries_match({CONF_HOST: netloc[0], CONF_PORT: int(netloc[1])})` before `get_status` so we can avoid connecting to the device every time an ssdp discovery packet is received when its already setup

Reply:
Good call, thanks! I will add that as well.